drivers/adc-dac/ad74416h/ad74416h.c

```diff
@@ -947,6 +966,12 @@ int ad74416h_remove(struct ad74416h_desc *desc)
 	if (!desc)
 		return -EINVAL;
 
+	if (desc->reset_gpio) {
+		ret = no_os_gpio_remove(desc->reset_gpio);
+		if (ret)
+			return ret;
+	}
```

Comment:
Also set desc->reset_gpio to NULL after removal

Reply:
This should be handled in the no_os_gpio_remove function, in my opinion.

Reply:
added #1894 addressing this issue.